core/src/main/java/com/amplitude/core/Amplitude.kt

```diff
@@ -70,29 +77,45 @@ open class Amplitude internal constructor(
         return Timeline().also { it.amplitude = this }
     }
 
-    open fun build(): Deferred<Boolean> {
-        storage = configuration.storageProvider.getStorage(this)
-        idContainer = IdentityContainer.getInstance(
-            IdentityConfiguration(
-                instanceName = configuration.instanceName,
-                apiKey = configuration.apiKey,
-                identityStorageProvider = IMIdentityStorageProvider(),
-                logger = configuration.loggerProvider.getLogger(this)
-            )
+    protected open fun createIdentityConfiguration(): IdentityConfiguration {
+        return IdentityConfiguration(
+            instanceName = configuration.instanceName,
+            apiKey = configuration.apiKey,
+            identityStorageProvider = configuration.identityStorageProvider,
+            logger = logger
         )
+    }
+
+    protected fun createIdentityContainer(identityConfiguration: IdentityConfiguration) {
+        idContainer = IdentityContainer.getInstance(identityConfiguration)
         val listener = AnalyticsIdentityListener(store)
         idContainer.identityManager.addIdentityListener(listener)
         if (idContainer.identityManager.isInitialized()) {
             listener.onIdentityChanged(idContainer.identityManager.getIdentity(), IdentityUpdateType.Initialized)
         }
+    }
+
+    private fun build(): Deferred<Boolean> {
```

Comment:
This used to be `open` maybe should we keep it that way for backwards compatiblity?

Reply:
`build()` is invoked in Amplitude constructor (`init`) and should not be invoked explicitly. Actually I had a error in tests when I used `build().await()` instead of `isBuilt.await()`
If someone currently invokes `build()` explicitly - it is definitely a bug, I believe.

Reply:
I was thinking more about overriding versus direct access to the method. What if we can make it `open protected` to allow overriding in subclasses but still prevent direct access?

Reply:
Fixed. `protected open` now